### DISCUSSION TITLE
Additional field or validator types for different kinds of numbers

### DISCUSSION THREAD

**Discussion by julianwachholz** [stefanfoulis/django-phonenumber-field#609] [Ideas]
I'd like to propose a potential feature to introduce new field classes and/or ready to use validator functions that could cover uses cases like enforcing a mobile phone number on some fields. (Or landline, toll-free etc.)

These identifiers exist upstream and the Django library could provide readily available model and form fields as well as validator functions.

One caveat I already know of is that the type of number cannot be determined by the number alone in all regions, so allowing unknown type would have to be configurable.

**Comment by francoisfreitag** [comment]
I think the validators should be enough, the model and form fields can be configured to use validators following Django patterns.
Because there are many variations one could want, I believe the following bar should be set for inclusion:
- the validator is generally useful, and
- the validator has a clear use case associated (e.g. no premium number), and
- the validator does what it says and has clear warnings about its limitations (e.g. a `mobile_phone_number` validator [will fail for the US region](https://github.com/daviddrysdale/python-phonenumbers/blob/43df39e674cfdf41024aaf9f0ec23cfc53938d98/python/phonenumbers/phonenumberutil.py#L462-L464)).

In case of doubts on the above, the validator should not be trivial. If it is, there is little value in keeping all variations of properties checks from the PhoneNumber object to match all possible use cases. An example of how simple a validator can be:
https://github.com/stefanfoulis/django-phonenumber-field/blob/f7765f4e420fc740a064f03c4c4f00db9162b7bd/phonenumber_field/validators.py#L8-L13
